### ISSUE TITLE
BareMetalHost stuck in provisioning/reboot loop, image install not working

### ISSUE THREAD

**Issue by fabrichter** [syself/cluster-api-provider-hetzner#802]
/kind bug

**What steps did you take and what happened:**
I followed the instructions on from [here](https://github.com/syself/cluster-api-provider-hetzner/blob/main/docs/topics/quickstart.md) to set up a cluster including bare metal worker nodes, using the `hetzner-hcloud-control-planes` flavor.
CNI and CCM (Hetzner Cloud Controller Manager) are set up as described there.
The control plane and worker nodes on hcloud came up successfully, but the `HetznerBareMetalHosts` couldn't be successfully provisioned. They are stuck in a loop of reboots and attempts to install the image.
Accessing the servers in question with SSH shows them either in the Hetzner rescue system in the process of installing the image, or in the previously installed system; the installimage script seems to have had no effect.

**What did you expect to happen:**
Bare metal hosts should be provisioned properly, installimage should install all the necessary components and the machines should then join the cluster.

**Anything else you would like to add:**
Is there a preparation step I could have missed or not done properly?
Credentials (for the Robot API and SSH) seem to work, the servers have been renamed in robot and are being rebooted via the API, and the rootDeviceHints field has been set.
I'll attach the output of ` kubectl get -n caph-clusters --ignore-not-found=true --watch --output-watch-events=true hbmh -o yaml`
The objects are in a loop of `registering -> image-installing -> provisioning`.
[hbmh-logs.yml.txt](https://github.com/syself/cluster-api-provider-hetzner/files/11432913/hbmh-logs.yml.txt)

**Environment:**

- cluster-api-provider-hetzner version: v1.0.0-beta.17
- Kubernetes version: (use `kubectl version`)
Client Version: v1.25.0
Kustomize Version: v4.5.7
Server Version: v1.25.3
- OS (e.g. from `/etc/os-release`): Ubuntu 20.04.5 LTS

**Comment by fabrichter** [comment]
Thanks! Here are the logs from `caph-controller-manager`.
[caph-controller-logs.txt](https://github.com/syself/cluster-api-provider-hetzner/files/11452829/caph-controller-logs.txt)

For context: The controller has access to one of the servers via SSH on the currently installed system because I manually ran install-image there to see that there were no problems with the script or machine. I used the SSH key we created for CAPH and also set the hostname to `bare-metal-test-1` as part of that.
On the other server, the SSH key we created for CAPH is not authorized in the currently installed system.

Do you have a way to get more logs from the install-image process? I think that might also be interesting.

**Comment by janiskemper** [comment]
This looks really weird to me. Our e2e tests do not have problems atm and installimage is usually stable (or at least should work if you try it again). Are you sure that you really configured the SSH keys correctly? 
Sometimes they are uploaded under the wrong name. Sometimes the wrong key is uploaded. Are you sure that this is not the case?

**Comment by fabrichter** [comment]
I'll create a new SSH key and try that under a different name.
I assumed SSH was working because the hardware details were populated by the operator. Is that how it works?

**Comment by fabrichter** [comment]
y

**Comment by janiskemper** [comment]
You have the ssh key for the standard os of the server and for the rescue system. Maybe something is wrongly configured there. While you specify the first one in the machine, the second is specified in the cluster.

**Comment by fabrichter** [comment]
I think the problem ended up being the `rootDeviceHints` configuration.
All the servers I tested have two disks built in which were previously configured as a software RAID-1.
I had used and tested that configuration previously for the BareMetalHost as well, but ended up using just one device (no RAID)  as part of my debugging. After going back to a RAID configuration, one of the servers (where I had previously run installimage manually) got successfully provisioned; another one was still stuck, but after manually running installimage there as well, it ended up succeeding for this one too.
